Create integration test suite for the Graph Database (#814)
hashintel/hash: packages/graph/hash_graph/README.md

```shell
cargo make doc --open
```

## Test the code

> The code base consist of two test suites: The unit test suite and the integration tests. To run the unit-test suite, simply run the `test` command:

Reviewer: ```suggestion
The code base has two test suites: The unit test suite and the integration tests. To run the unit-test suite, simply run the `test` command:
```

It's made-up of more than just tests 😁
